src/main/java/com/ksabbak/javaserver/storage/Store.java

```diff
@@ -0,0 +1,47 @@
+package com.ksabbak.javaserver.storage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Store implements Persistable {
+    private Map<String, String> store;
+
+    public Store(){
+        store = new HashMap<String, String>();
+    }
+
+    @Override
+    synchronized public String create(String key) throws Exception {
+        return create(key, "");
+    }
+
+    @Override
+    synchronized public String create(String key, String value) throws Exception {
+        if (store.get(key) != null) throw new Exception();
+        store.put(key, value);
+        return key;
+    }
+
+    @Override
+    synchronized public String update(String key, String value) throws Exception{
+        if (store.get(key) == null) throw new Exception();
+        store.put(key, value);
+        return key;
+    }
+
+    @Override
+    public String read(String key) throws Exception{
+        String value = store.get(key);
+        if (value == null) throw new Exception();
+        return value;
+    }
+
+    @Override
+    synchronized public String delete(String key) throws Exception{
+        String value = store.remove(key);
+        if (value == null){
```

Comment:
`delete` feels to me like it could be idempotent, and just do nothing if the key does not exists. Admittedly, it is consistent with the other methods.

Reply:
I like the consistency and also while ultimately deleting something that's there and not deleting something that isn't there has the same results, I don't want to risk the idea that there might be some cases where the action of deleting is part of what's relevant. Like if a successful deletion triggers an email or something?